### ISSUE TITLE
DBMS remote url

### ISSUE THREAD

**Issue by masoom80** [app-generator/devtool-python-converter#26]
Hi!
could you explain to me how this part should be done?

**Comment by app-generator** [comment]
Hello @masoom80 

This is the use case:
The user provides the connection URL:
- ip
- port
- dbname
- pass
- DB driver

and the converter should execute a reflection on the database and extract tables. 

A tool that already does this is here:

https://github.com/app-generator/devtool-db-introspection

[Peewee](http://docs.peewee-orm.com/en/latest/) is used to implement (almost) all the above requirements. 

The goal of this feature is: 

- detect a remote DB structure (all tables)
- user selects a table
- From this point, he should be able to:
  - covert model to FLask & Django definition
  - import data into a DataTable 
  - export .. 

Basically, we can associate a table + data to a CSV file. 

CSV Colums == Table Columns
CVS Data == Table Data  

All conversions applied to a CSV file should be also applicable to a remote table. 

Let me know if you need more information. 

Ty!

**Comment by masoom80** [comment]
Hi @app-generator 
Should this part be done in memory or is file-based processing okay?
and should I put a limit on the size of the table requested by the user?

**Comment by masoom80** [comment]
@app-generator 
Can you give me an input samlpe so i can test my implementation?

**Comment by app-generator** [comment]
hello @masoom80 

Yep, all processing should be done `in memory`, without any file or local storage. Once all features are stable, the code is moved much as possible in the library that can be used in various contexts:

- web apps (the main execution context)
- stand-alone projects 
- CLI tools 

Being in-memory, the code can be used in all the above contexts without touching the core. 

For further Q, I'm here. 

Ty!

**Comment by app-generator** [comment]
P.S. The processing limit (all inputs) is saved in the configuration:

- CFG variable:  `INPUT_LIMIT`
  - Default value: `500` (rows)

All rows above this value are ignored.

**Comment by app-generator** [comment]
Hello @masoom80 

We don't have a public DB URL but feel free to create a simple SQLite file from an ordinary project and try to process it. 
If you have issues with this part, just use this Flask sample that generates a minimal SQLite database at first use: 

https://github.com/app-generator/flask-material-kit

Let me know if you have further Q. For sure, I can help. 

ty!

**Comment by alivarastepour** [comment]
> Hello @masoom80
> 
> We don't have a public DB URL but feel free to create a simple SQLite file from an ordinary project and try to process it. If you have issues with this part, just use this Flask sample that generates a minimal SQLite database at first use:
> 
> https://github.com/app-generator/flask-material-kit
> 
> Let me know if you have further Q. For sure, I can help.
> 
> ty!

hi @app-generator 
could you please give me the format in which connection URL is provided?

**Comment by app-generator** [comment]
Hello guys,

The input UI/UX for remote resources (CSV, DB) is detailed here:

https://github.com/app-generator/devtool-python-converter/issues/27#issuecomment-1365737058

Basically, the DB connection string is a composition of inputs provided by the user:

- DB Engine
- HOST (i, pr hostname) / PORT
- Credentials: user/pass

We cannot ask directly for the connection string because the syntax might lead to `TYPO` errors. 

Let me know if you have further Q. 

TY!

**Comment by masoom80** [comment]
Hi @app-generator 
We have implemented this part, please provide us with a  public DB URL to do the final tests.

**Comment by masoom80** [comment]
Hi again @app-generator 
Could you give us the link?
we can't fully test the project without it.
ty!

> Hi @app-generator We have implemented this part, please provide us with a public DB URL to do the final tests.

**Comment by app-generator** [comment]
Hello @masoom80,

Here is a source for SQLite: 

https://github.com/app-generator/sample-flask-sqlite-source/blob/master/apps/db.sqlite3

For testing the DBMS like MySQL .. etc, please use any local database you have.

[Peewee](http://docs.peewee-orm.com/en/latest/) supports:

- sqlite 
- mysql 
- PostgreSQL

Regarding the DBMS UI, I will open a new issue related to the UX (nothing fancy, just to be more strict with the input provided by the user). 

Ty!

**Comment by app-generator** [comment]
The UI looks great.
Ty guys!
